src/components/Upload.tsx

```diff
@@ -79,7 +82,12 @@ export default function Upload({ setInUploadView }: UploadProps) {
   const zipAndUpload = () => {
     setShowProgress(true);
     const zip = new JSZip();
-    files.map((file, index) => zip.file(`${index}.dcm`, file));
+    const ct = zip.folder("ct");
+    const rtdose = zip.folder("rtdose");
+    const rtstruct = zip.folder("rtstruct");
+    ctFiles.map((file, index) => ct?.file(`${index}.dcm`, file));
+    rtdoseFiles.map((file, index) => rtdose?.file(`${index}.dcm`, file));
+    rtstructFiles.map((file, index) => rtstruct?.file(`${index}.dcm`, file));
```

Comment:
Te 6 linijek to tak naprawdę dwie linijki które zostały powtórzone 3 razy. 
Prze to trochę trudniej to się czyta.
Ja bym zamiast tego, zrobiłbym to jako funkcję:
```js
function packFilesToFolders(folderName: string, files: File[]) {
   const folder = zip.folder(folderName);
   return files..map((file, idx) => folder.file(`${idx}.dcm`, file))
}
_.zip(['ct', 'rtDose', 'rtStruct'], [ctFiles, rtDoseFiles, rtStructFiles]).map(([fName, file]) => packFilesToFolders(fName, file)
```

Albo coś w tym stylu :) 

PS
`_` - to dość popularna biblioteka [`lodash`](https://lodash.com/)

Reply:
Przyjmując, że będziemy reprezentować stan jako obiekt, można tę ostatnią linijkę uprościć:

```js
Object.entries(state).map([fName, files] => ....)
```